packages/canon/src/components/Link/types.ts

```diff
@@ -30,4 +30,5 @@ export interface LinkProps
   weight?: 'regular' | 'bold' | Partial<Record<Breakpoint, 'regular' | 'bold'>>;
   className?: string;
   style?: CSSProperties;
+  render?: (props: Omit<LinkProps, 'render'>) => ReactNode;
```

Comment:
I think we wanna make it `component` or such. render is a bit of an older pattern afaik and isn't as explicit about whether it's ok to call hooks in the provided function

Reply:
This is to follow what Base UI is doing. I had the same impression as you but I would prefer a consistent pattern rather than two different behaviour.

https://base-ui.com/react/handbook/composition

Reply:
wattafakk, that looks a bit different though with the intended use being to pass an element with some constraints on the underlying component implementation.

I have to assume that Base UI is doing some trickery and cloning or re-rendering with the component type or something like that. I agree we should stick to what Base UI does though, including the ability to pass am element.

Still feel this leaves the confusion around whether it's alright or not to use hooks in the render function though, at this point I assume not 🤷